obspy/core/stream.py

```diff
@@ -3118,6 +3119,80 @@ def remove_sensitivity(self, *args, **kwargs):
             tr.remove_sensitivity(*args, **kwargs)
         return self
 
+    def stack(self, group='all', type='linear', npts_tol=0):
+        """
+        Return stream with traces stacked by the same selected metadata.
+
+        :param str group: Stack waveforms together which have the same metadata
```

Comment:
```suggestion
        :param group: Stack waveforms together which have the same metadata
```

Reply:
Just added this, because @d-chambers asked to document the types. For me, it does not matter.

Reply:
Then you have to use `:type group: str` as you already did for one of the other parameters. I'm not sure if sphinx can use the syntax you originally used but we probably stick to a consistent docstyle syntax.

Reply:
@krischer can you clarify a bit. IMHO it is not necessary to document the basic types, but it also does not hurt. Do you also want me to remove the `int` from `int npts_tol`?

Reply:
For the `group` parameter I would document the types as it is not obvious and it can take a string or a tuple. In general I also think its nice to be consistent within a single docstring and if a single type is documented, all should be documented.

But its all a bit moot as we'll hopefully drop Python 2 support very soon and can then just use type hints for documentation so we'll have some work to do in any case.

Reply:
Yeah sphinx is OK with this syntax, I use it all the time, because the other form is a bit verbose, but I can of course change it.